### PR TITLE
feat: emotes v2

### LINKED ISSUE #145: Emotes v2 (Godot)

### PR DESCRIPTION
**Pull request by leanmendoza**
Fix #145 

Support non-default emotes, and emotes V2 (with audio and extra meshes)

# What?
- Separate gltf loading pipeline with `content_type`
  - 1=Wearable doesn't generate colliders
  - 2=Emotes doesn't generate colliders and prepare emote animations
  - _=Scene Content, it generates colliders
- Bundle default emotes, shared resource reference between avatars
- Separate animation libraries, now, default emotes and locomotion are immutable shared resource
- Idle animation is unique resource per avatar, it's cloned in the constructor. The tracks on it depends on equipped emotes

### CODE REVIEW

File: godot/src/decentraland_components/avatar.gd [hunk 1]
```diff
@@ -71,6 +116,22 @@ func async_update_avatar_from_profile(profile: DclUserProfile):
 
 
 func async_update_avatar(new_avatar: DclAvatarWireFormat):
+	var emotes = PackedStringArray(
```
Comment (kuruk-mm) [inline godot/src/decentraland_components/avatar.gd:119]:
```suggestion
	# TODO: Hardcoded emotes, change those with the configured emotes
	var emotes = PackedStringArray(
```

File: godot/src/ui/components/discover/carrousel/discover_carrousel_item.gd [hunk 2]
```diff
@@ -54,11 +54,13 @@ func set_image(_texture: Texture2D):
 
 
 func set_title(_title: String):
-	label_title.text = _title
+	if is_instance_valid(label_title):
```
Comment (kuruk-mm) [inline godot/src/ui/components/discover/carrousel/discover_carrousel_item.gd:57]: what was the issue here?

File: godot/src/ui/components/emotes/emote_wheel.gd [hunk 3]
```diff
@@ -37,12 +39,63 @@ func _ready():
...
+		# get_emotes() always returns 10 emotes, but just in case
+		if i >= emote_urns.size():
+			# Set default or
+			continue
+
+		var emote_item := emote_items[i]
+		emote_item.emote_id = emote_urns[i]
+		emote_item.number = i
+
+		if is_emote_default(emote_item.emote_id):
```
Comment (kuruk-mm) [inline godot/src/ui/components/emotes/emote_wheel.gd:60]:
```suggestion
		# TODO: Set emote_name
		if is_emote_default(emote_item.emote_id):
```

File: rust/decentraland-godot-lib/src/avatars/item.rs [hunk 4]
```diff
@@ -83,49 +105,119 @@ impl DclWearableEntityDefinition {
...
+    #[func]
+    fn get_emote_loop(&self) -> bool {
+        let Some(emote_data) = self.inner.item.emote_data.as_ref() else {
+            return false;
+        };
+
+        emote_data.r#loop
+    }
+
+    /// Returns a 12 character long prefix for the emote id, removing the tokenId if it presents
```
Comment (kuruk-mm) [inline rust/decentraland-godot-lib/src/avatars/item.rs:132]: using 3 `///` usually is for a documentation comment, and I get warnings in the cargo run

we should start using double //

**Review by kuruk-mm — APPROVED** [review]
This is great!! :smile:

File: godot/src/ui/components/emotes/emote_wheel.gd (hunk 3 above)
Comment (leanmendoza) [reply godot/src/ui/components/emotes/emote_wheel.gd]: fixed

**Pull request merged**